### DISCUSSION TITLE
Refresh Breadcrumbs index through Templater js api

### DISCUSSION THREAD

**Discussion by HyperEpsilon** [SkepticMystic/breadcrumbs#473] [Q&A] [answered]
Hi, 

I'm creating a template through templater, and I was wondering if it was possible to refresh the breadcrumbs index via a javascript api.

I'm already using an execution template, so it would be very easy for me to add in a call to reload the breadcrumbs index, should such a function exist.

**Comment by SkepticMystic** [comment, marked as answer]
I think you're in luck, @HyperEpsilon.
Checkout the docs for the Breadcrumbs API, specifically this function: https://skepticmystic.github.io/breadcrumbs/interfaces/BCAPII.html#refreshIndex

**Reply by HyperEpsilon** [reply]
Ah wonderful, that's exactly what I was looking for.

When creating the API link, what is the difference between app.plugins.plugins.breadcrumbs.api vs window.BCAPI?

**Reply by SkepticMystic** [reply]
I don't think there is much practical difference.
`window.BCAPI` is there for convenience of access. You can technically use `BCAPI` at the top level, as well.

**Reply by HyperEpsilon** [reply]
Alrighty, I got things working after a couple of tries. I had to put the refresh in a separate async call, but now everything works.  
I tried differing delays, and 500 ms seems to be a good middle ground to let the system catch up before attempting to refresh.

I included this snippet at the bottom of my templater template:
```js
<%* 
async function refreshBC() {
  await new Promise(resolve => setTimeout(resolve, 500));
  await window.BCAPI.refreshIndex(); 
}

refreshBC();
%>
```

**Comment by michaelpporter** [comment]
Closing as resolved — the Templater/index API changed substantially in v4. Reopen if you still need this against the current version.
